### PR TITLE
Allow users to increase MAX_MONEY

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
# Description
Screenshot of the Trainer Card with max money

* Allows the user to increase `MAX_MONEY` to any number up to `INT_MAX`, while menus and texts will adjust to match.
* Refactors `CountDigits` to work with values greater than 100M.

# Usage

## `MAX_MONEY`
In [`include/money.h`](https://github.com/PokemonSanFran/pokeemerald-expansion/blob/upcoming/include/money.h), developers can change `MAX_MONEY` to any integer less than `2147483648`.

# Testing
## Clean Branch
You can recreate this branch by applying a patch or pulling the repo. From a clean version of expansion's upcoming, you can either:

### Patch
`wget https://files.catbox.moe/1urr6w.patch -O increaseMoney.patch ; git apply increaseMoney.patch ; rm increaseMoney.patch`

### Repo
`git remote add psf-expansion https://github.com/PokemonSanFran/pokeemerald-expansion/ ; git pull psf-expansion increaseMoney`

## Manual Tests
After replicating the branch, to recreate my testing environment, you can either directly download the debug script and config file, or manually create the changes.

### Download
| `MAX_MONEY`  |  Command |
|---|---|
|`9`|`wget https://files.catbox.moe/ourb5p.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`99`|`wget https://files.catbox.moe/pbkxuj.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`999`|`wget https://files.catbox.moe/l0gyte.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`9999`|`wget https://files.catbox.moe/l1n9g8.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`99999`|`wget https://files.catbox.moe/2b4bw0.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`999999`|`wget https://files.catbox.moe/e831bv.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`9999999`|`wget https://files.catbox.moe/4ehple.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`99999999`|`wget https://files.catbox.moe/9yhna4.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`999999999`|`wget https://files.catbox.moe/g5yanb.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|
|`INT_MAX`|`wget https://files.catbox.moe/f82nww.h -O include/money.h && wget https://files.catbox.moe/0kor2h.h -O src/data/items.h && wget https://files.catbox.moe/wsd3z6.inc -O data/scripts/debug.inc`|

### Manual Testing
* Change `MAX_MONEY` to its desired value in [`include/money.h`](https://github.com/PokemonSanFran/pokeemerald-expansion/blob/upcoming/include/money.h)
* Change [`data/scripts/debug.inc`](https://files.catbox.moe/wsd3z6.inc) to match the one provided
* Change [`src/data/items.h`](https://files.catbox.moe/0kor2h.h) to match the one provided
* Compile the game
* Start a new save
* Run Debug Script 1
* Buy and sell items

## Verified Scenarios

All videos show the player:
* opens and then closes Trainer Card
* runs Debug Script 1
* talks to the Ever Grande shopkeeper
* sells 4 Big Nugget
* sells 1 Potion
* buys 1 Ultra Ball
* opens Trainer Card

### `MAX_MONEY == 9`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/cfb01520-c3a9-4aa3-a02f-6f48a2856999
### `MAX_MONEY == 99`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/8be6f8ae-8cef-4f78-87e5-5f2272daab95
### `MAX_MONEY == 999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/c1b7c40d-afa6-413d-b6d5-71513cade1ff
### `MAX_MONEY == 9999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/4623230d-6204-412e-8f29-f1d4a2c1438b
### `MAX_MONEY == 99999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/114554ea-fa6e-496c-9131-d56aca3fe779
### `MAX_MONEY == 999999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/8a4af931-de4f-48dd-8cd8-29a0065a3b1b
### `MAX_MONEY == 9999999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/af08f286-fba7-4a68-9d98-4cf4577c6cda
### `MAX_MONEY == 99999999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/5eccd14f-575e-4a18-952a-f95b2b3fec19
### `MAX_MONEY == 999999999`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/fce5221b-6d25-43dd-a533-1387033cd8d0
### `MAX_MONEY == INT_MAX`
https://github.com/rh-hideout/pokeemerald-expansion/assets/77138753/0312dea5-d497-4812-b7cf-1bd83e2bbe13

# **People who collaborated with me in this PR**
This feature was [originally written by](https://github.com/pret/pokeemerald/wiki/Increase-money-limit) @DizzyEggg.

# Features this PR does NOT handle:
n/a

# Discord Contact Info
I am `pkmnsnfrn` on Discord.

### CODE REVIEW

**Comment by pkmnsnfrn** [comment]
Explicitly requesting the eyes of @DizzyEggg _and_ @mrgriffin , and I'm happy to wait until they have time or they refuse.

**Comment by Bassoonian** [comment]
You have tabs instead of spaces in the vast majority of your files

**Comment by pkmnsnfrn** [comment]
> You have tabs instead of spaces in the vast majority of your files
~~FUCK~~

https://github.com/rh-hideout/pokeemerald-expansion/pull/4943/commits/f1f285e612fb1702115de084b437a8e3d3552e3d

**Review by Bassoonian — APPROVED** [review]

File: src/money.c [hunk 1]
```diff
@@ -130,23 +131,33 @@ void SubtractMoneyFromVar0x8005(void)
...
+    leadingSpaces = CalculateLeadingSpacesForMoney(numDigits);
 
-    while (strLength-- > 0)
+    while (leadingSpaces-- > 0)
         *(txtPtr++) = CHAR_SPACER;
 
     StringExpandPlaceholders(txtPtr, gText_PokedollarVar1);
+
+    if (numDigits > 8)
+        PrependFontIdToFit(gStringVar4, txtPtr + 1 + numDigits, FONT_NORMAL, 54);
```
Comment (mrgriffin) [inline src/money.c:160]: Initially I was going to say that `PrependFontIdToFit` should work for any number of digits (it just doesn't do anything if the text already fits), but perhaps I'm wrong due to the alignment?

**Review by mrgriffin — COMMENTED** [review]
Nice to see shrinking fonts being used! I think you've summoned me to comment on those bits, and I don't see anything wrong :)

File: src/money.c (hunk 1 above)
Comment (pkmnsnfrn) [reply src/money.c:160]: Yeah the spaces fuck it all up, and when I remove the spaces and try to manually align, the text doesn't get overwritten if you change the number of digits mid transaction

**Pull request merged**